### PR TITLE
Update ACL tests and modify JedisPool tests

### PR DESCRIPTION
**Pull request by sazzad16**
- Test ACL with non-default user for standalone Redis instance (Jedis, Jedis with SSL, JedisPool)
- Removed Jedis with ACL tests where ACL feature was not being used and will not bring anything new even if used
- Removed JedisPool with ACL tests where ACL feature was not being used and will not bring anything new even if used
- Modify JedisPool (with/without ACL) tests
- Used try-with-resources in JedisPool (with/without ACL) tests
- Added a test related to issue #2318 and PR #2319 
- Added two minimal JedisPool constructor with user and password params (also testing those)

### CODE REVIEW

File: src/test/java/redis/clients/jedis/tests/JedisPoolTest.java [hunk 1]
```diff
@@ -42,56 +40,64 @@ public void checkConnections() {
   @Test
   public void checkCloseableConnections() throws Exception {
     JedisPool pool = new JedisPool(new JedisPoolConfig(), hnp.getHost(), hnp.getPort(), 2000);
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/JedisPoolTest.java:42]: @sazzad16 I think the JedisPool is also closable

File: src/test/java/redis/clients/jedis/tests/JedisPoolTest.java [hunk 2]
```diff
@@ -42,56 +40,64 @@ public void checkConnections() {
...
-  public void checkJedisIsReusedWhenReturned() {
+  public void checkResourceIsClosableAndReusable() {
+    GenericObjectPoolConfig config = new GenericObjectPoolConfig();
+    config.setMaxTotal(1);
+    config.setBlockWhenExhausted(false);
+    JedisPool pool = new JedisPool(config, hnp.getHost(), hnp.getPort(), 2000, "foobared", 0,
+        "closable-resuable-pool", false, null, null, null);
 
-    JedisPool pool = new JedisPool(new JedisPoolConfig(), hnp.getHost(), hnp.getPort());
     Jedis jedis = pool.getResource();
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/JedisPoolTest.java:72]: Jedis is closeable

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/test/java/redis/clients/jedis/tests/JedisPoolTest.java (hunk 1 above)
Comment (sazzad16) [reply src/test/java/redis/clients/jedis/tests/JedisPoolTest.java:42]: @gkorland we need for `assertTrue(pool.isClosed());` below

File: src/test/java/redis/clients/jedis/tests/JedisPoolTest.java (hunk 2 above)
Comment (sazzad16) [reply src/test/java/redis/clients/jedis/tests/JedisPoolTest.java]: @gkorland https://github.com/redis/jedis/pull/2325/commits/b9bf1acbab6b9ac9f43813b686d7bacdf5feb097

**Review by gkorland — APPROVED** [review]

**Pull request merged**